src/config/reader.rs

```diff
@@ -117,7 +117,7 @@ impl ConfigReader {
                 }
                 LinkType::Protobuf => {
                     let descriptors = self
-                        .resolve_descriptors(HashMap::new(), source.path)
+                        .resolve_descriptors(HashMap::new(), self.read_proto(&source.path).await?)
```

Comment:
not related to changes, but why do we pass `HashMap` to the call instead of creating it inside the `resolve_descriptors` itself?

Reply:
earlier the implementation was different, someone refactored it and now the function is returning a hashmap, we don't need to take hashmap as arg. I'll raise separate pr for later